src/cryptsetup/cryptsetup.c

```diff
@@ -2373,11 +2420,30 @@ static int run(int argc, char *argv[]) {
 
 /* since cryptsetup 2.7.0 (Jan 2024) */
 #if HAVE_CRYPT_SET_KEYRING_TO_LINK
-                        if (arg_link_key_description) {
+                        if (arg_link_keyring && arg_link_key_description)
                                 r = crypt_set_keyring_to_link(cd, arg_link_key_description, NULL, arg_link_key_type, arg_link_keyring);
-                                if (r < 0)
-                                        log_warning_errno(r, "Failed to set keyring or key description to link volume key in, ignoring: %m");
+                        else if (arg_link_keyring) {
+                                _cleanup_free_ char *key_description_1 = NULL, *key_description_2 = NULL;
+
+                                key_description_1 = strjoin("systemd-cryptsetup:vk-",
+                                                            strempty(crypt_get_uuid(cd)),
+                                                            "-id1");
```

Comment:
hmm, why suffix this with -id1? i mean, it would suffice to suffix the 2nd vk like this?